File: server/src/lobby/packet/packet.js
const PACKET_ID = require("./pakcet-list.json");

class Packet {
  constructor(protocol) {
    this.Protocol = protocol;
  }
}

class C_CreateRoom extends Packet {
  userName;

  constructor() {
    super(PACKET_ID.C_CreateRoom);
  }
}

class C_EnterRoom extends Packet {
  userName;
  roomId;

  constructor() {
    super(PACKET_ID.C_EnterRoom);
  }
}

class S_CreateRoom extends Packet {
  user;
  roomId;

  constructor() {
    super(PACKET_ID.S_CreateRoom);
  }
}

class S_EnterRoom extends Packet {
  user;
  roomId;

  constructor() {
    super(PACKET_ID.S_EnterRoom);
  }
}

class S_LeaveRoom extends Packet {
  user;

  constructor() {
    super(PACKET_ID.S_LeaveRoom);
  }
}

class S_UserList extends Packet {
  users = [];

  constructor() {
    super(PACKET_ID.S_UserList);
  }
}

module.exports = {
  ID: PACKET_ID,
  C_CreateRoom,
  C_EnterRoom,
  S_CreateRoom,
  S_EnterRoom,
  S_LeaveRoom,
  S_UserList,
};

File: human-or-not/Assets/Scripts/Contents/Network/Packet/Packet.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public enum PacketId
{
    // S: Server's packet, C: Client's packet
    C_CreateRoom = 1,
    C_EnterRoom = 2,
    S_CreateRoom = 11,
    S_EnterRoom = 12,
    S_LeaveRoom = 13,
    S_UserList = 14,
}

public abstract class Packet
{
    public ushort Protocol;
}

#region Interface
[Serializable]
public class UserInfo
{
    public string id;
    public string name;
}
#endregion

#region Server's packet
[Serializable]
public class S_CreateRoom : Packet
{
    public UserInfo user;
    public string roomId;
}

[Serializable]
public class S_EnterRoom : Packet
{
    public UserInfo user;
    public string roomId;
}

[Serializable]
public class S_LeaveRoom : Packet
{
    public UserInfo user;
}

[Serializable]
public class S_UserList : Packet
{
    public UserInfo[] users;
}
#endregion

#region Client's packet
[Serializable]
public class C_CreateRoom : Packet
{
    public string userName;
    public C_CreateRoom()
    {
        Protocol = (ushort)PacketId.C_CreateRoom;
    }
}

[Serializable]
public class C_EnterRoom : Packet
{
    public string userName;
    public string roomId;

    public C_EnterRoom()
    {
        Protocol = (ushort)PacketId.C_EnterRoom;
    }
}
#endregion